crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1270,29 +1307,30 @@ impl<'db> Type<'db> {
         let iterable_meta_type = self.to_meta_type(db);
 
         let dunder_iter_method = iterable_meta_type.member(db, "__iter__");
-        if let Symbol::Type(dunder_iter_method, _) = dunder_iter_method {
+        if let Symbol::Type(dunder_iter_method, boundness) = dunder_iter_method {
             let Some(iterator_ty) = dunder_iter_method.call(db, &[self]).return_ty(db) else {
```

Comment:
Should/can we use `call_dunder` here?

Reply:
It's slightly awkward since we only use the boundness information inside the nested conditional, but it's only slightly more code — while being more consistent.

We can actually also use it below for `__get_item__` where it improved the code a lot.